tests/chainer_tests/iterators_tests/test_shuffled_iterator.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+
+from chainer import iterators
+from chainer import testing
+
+
+class TestShuffledIterator(unittest.TestCase):
+
+    def test_iterator_repeat(self):
+        dataset = [1, 2, 3, 4, 5, 6]
+        it = iterators.ShuffledIterator(dataset, 2)
+        for i in range(3):
+            self.assertEqual(it.epoch, i)
+            batch1 = it.next()[:]
+            self.assertEqual(len(batch1), 2)
+            self.assertFalse(it.is_new_epoch)
+            batch2 = it.next()[:]
+            self.assertEqual(len(batch2), 2)
+            self.assertFalse(it.is_new_epoch)
+            batch3 = it.next()[:]
+            self.assertEqual(len(batch3), 2)
+            self.assertTrue(it.is_new_epoch)
+            self.assertEqual(sorted(batch1 + batch2 + batch3), dataset)
```

Comment:
How about adding unique operation so that we can ensure there is no duplication.